### PR TITLE
feat: [PL-22071]: add enforcement banner

### PR DESCRIPTION
**Pull request by vkbansal**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by kapilkumawat86 — DISMISSED** [review]

File: src/framework/featureStore/FeaturesFactory.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import type { FeatureIdentifier } from './FeatureIdentifier'
+import type { CheckFeaturesReturn } from './featureStoreUtil'
+
+export interface FeatureProps {
+  features: FeatureIdentifier[]
+  renderMessage(props: CheckFeaturesReturn): React.ReactNode | null
+}
+
+export type Module = 'ci' | 'cd' | 'cf' | 'cv' | 'ce'
```
Comment (abhinavrastogi-harness) [inline src/framework/featureStore/FeaturesFactory.ts:9]: dont we already have the Module type in common interfaces?

File: src/modules/10-common/interfaces/RouteInterfaces.ts [hunk 2]
```diff
@@ -134,17 +134,11 @@ export interface ServicePathProps {
...
-  | 'cd'
-  | 'cf'
-  | 'cv'
-  | 'ce'
-  | ':module(ci)'
-  | ':module(cd)'
-  | ':module(cf)'
-  | ':module'
-  | ':module(cv)'
+export type ModuleName = 'ci' | 'cd' | 'cf' | 'cv' | 'ce'
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/interfaces/RouteInterfaces.ts:137]: cant we reused this across files?

File: src/framework/featureStore/FeaturesFactory.ts (hunk 1 above)
Comment (vkbansal) [reply src/framework/featureStore/FeaturesFactory.ts:9]: I didn't do it because I didn't want to import from modules into framework

**Review by chiragp1212 — DISMISSED** [review]

File: src/modules/10-common/interfaces/RouteInterfaces.ts (hunk 2 above)
Comment (vkbansal) [reply src/modules/10-common/interfaces/RouteInterfaces.ts]: Fixed it

**Review by kapilkumawat86 — APPROVED** [review]

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**